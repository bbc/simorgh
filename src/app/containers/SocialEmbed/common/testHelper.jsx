import React from 'react';

import { RequestContextProvider } from '#contexts/RequestContext';
import { ServiceContextProvider } from '#contexts/ServiceContext';
import { STORY_PAGE } from '#app/routes/utils/pageTypes';

/**
 * withContexts is a higher-order component that returns a React component
 * wrapped in the necessary page contexts for testing and Stories.
 * @param {Function} Component A React component.
 * @param {Object} Configuration A configuration object.
 */
const withContexts =
  (Component, { isAmp, service = 'news' }) =>
  props =>
    (
      <RequestContextProvider
        isAmp={isAmp}
        pageType={STORY_PAGE}
        service={service}
        pathname="/pathname"
      >
        <ServiceContextProvider service={service}>
          <Component {...props} />
        </ServiceContextProvider>
      </RequestContextProvider>
    );

export default withContexts;
