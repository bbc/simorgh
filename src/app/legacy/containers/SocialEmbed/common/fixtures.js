export const cpsTwitterBlockNoEmbed = {
  type: 'twitter',
  indexOfType: 0,
  model: {
    href: 'https://twitter.com/MileyCyrus/status/1237210910835392512',
    id: '1237210910835392512',
  },
  id: 'd5bccd4e-60ae-471d-adcc-9de0f9a6c1c7',
  position: [27, 1],
};

export const cpsTwitterBlock = {
  type: 'twitter',
  indexOfType: 1,
  model: {
    href: 'https://twitter.com/MileyCyrus/status/1237210910835392512',
    id: '1237210910835392512',
    embed: {
      oembed: {
        indexOfType: 1,
        url: 'https://twitter.com/MileyCyrus/status/1237210910835392512',
        author_name: 'Miley Ray Cyrus',
        author_url: 'https://twitter.com/MileyCyrus',
        html: '<blockquote class="twitter-tweet"><p lang="en" dir="ltr">Australia: Due to the recommendations of local, state, federal and international government authorities, including the Center for Disease Control, to reduce potential health risks in response to the current global health crisis, we are no longer traveling to Aus for the show.</p>&mdash; Miley Ray Cyrus (@MileyCyrus) <a href="https://twitter.com/MileyCyrus/status/1237210910835392512?ref_src=twsrc%5Etfw">March 10, 2020</a></blockquote>\n',
        width: 550,
        height: null,
        type: 'rich',
        cache_age: '3153600000',
        provider_name: 'Twitter',
        provider_url: 'https://twitter.com',
        version: '1.0',
      },
      fallback_image: {
        fallback_image_width: 465,
        fallback_image_height: 279,
        alt_text:
          'Twitter post by Miley Ray Cyrus: Australia: Due to the recommendations of local, state, federal and international government authorities, including the Center for Disease Control, to reduce potential health risks in response to the current global health crisis, we are no longer traveling to Aus for the show.',
      },
    },
  },
  id: 'c063a42a-53f9-4421-ba8d-db7ffbd2bfe6',
  position: [25, 1],
};

export const twitterBlock = {
  type: 'renditions',
  model: {
    locator: 'urn:bbc:optimo:social:2777c9b9-b3d3-4d53-b36c-be749aa528c9',
    blocks: [
      {
        type: 'aresOEmbed',
        model: {
          oembed: {
            indexOfType: 0,
            version: '1.0',
            author_name: 'BBC News (UK)',
            author_url: 'https://twitter.com/BBCNews',
            provider_name: 'Twitter',
            provider_url: 'https://twitter.com',
            url: 'https://twitter.com/BBCNews/status/1384138850478346243',
            html: '<blockquote class="twitter-tweet"><p lang="en" dir="ltr">Greta Thunberg says meeting fellow climate campaigner Sir David Attenborough was &quot;indescribable&quot; <a href="https://t.co/xz93WmAdfR">https://t.co/xz93WmAdfR</a></p>&mdash; BBC News (UK) (@BBCNews) <a href="https://twitter.com/BBCNews/status/1384138850478346243?ref_src=twsrc%5Etfw">April 19, 2021</a></blockquote>\n',
            width: 550,
          },
        },
      },
    ],
  },
};

export const twitterBlockNoEmbed = {
  type: 'renditions',
  model: {
    locator: 'urn:bbc:optimo:social:2777c9b9-b3d3-4d53-b36c-be749aa528c9',
    blocks: [
      {
        type: 'aresOEmbed',
      },
    ],
  },
};

export const instagramBlock = {
  type: 'renditions',
  model: {
    locator: 'urn:bbc:optimo:social:2777c9b9-b3d3-4d53-b36c-be749aa528c9',
    blocks: [
      {
        type: 'aresOEmbed',
        model: {
          oembed: {
            indexOfType: 0,
            version: '1.0',
            author_name: 'BBC Turkce',
            author_url: 'https://www.instagram.com/bbcturkce/',
            provider_name: 'Instagram',
            provider_url: 'https://instagram.com',
            url: 'https://www.instagram.com/reel/CeWO3HcIE9w/?utm_source=ig_embed&ig_rid=b6b91062-7174-4784-9a99-139d52bc5b29',
            html: '<blockquote class="instagram-media" data-instgrm-captioned data-instgrm-permalink="https://www.instagram.com/p/CgNAEjOK46_"><p><a href="https://www.instagram.com/p/CgNAEjOK46_" style="color:#000">Â© 2020 Instagram</a></p></blockquote> <script async src="https://www.instagram.com/embed.js"></script>',
            width: 550,
          },
        },
      },
    ],
  },
};

export const instagramBlockNoEmbed = {
  type: 'renditions',
  model: {
    locator: 'urn:bbc:optimo:social:2777c9b9-b3d3-4d53-b36c-be749aa528c9',
    blocks: [
      {
        type: 'aresOEmbed',
      },
    ],
  },
};

export const youtubeBlockEmbed = {
  type: 'renditions',
  model: {
    locator: 'urn:bbc:optimo:social:d4ae3e6f-51cf-4899-bbd4-16d14882cbc7',
    blocks: [
      {
        type: 'aresOEmbed',
        model: {
          oembed: {
            title:
              'Rick Astley - Never Gonna Give You Up (Official Music Video)',
            author_name: 'Rick Astley',
            author_url: 'https://www.youtube.com/user/BBCMundo',
            type: 'video',
            height: 113,
            width: 200,
            version: '1.0',
            provider_name: 'YouTube',
            provider_url: 'https://www.youtube.com/',
            thumbnail_height: 360,
            thumbnail_width: 480,
            thumbnail_url: 'https://i.ytimg.com/vi/chiWVxreqhU/hqdefault.jpg',
            html: '<iframe width="200" height="113" src="https://www.youtube.com/embed/1e05_rwHvOM?feature=oembed" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>',
          },
        },
      },
    ],
  },
};

export const tiktokBlockEmbed = {
  type: 'renditions',
  model: {
    locator: 'urn:bbc:optimo:social:d4ae3e6f-51cf-4899-bbd4-16d14882cbc7',
    blocks: [
      {
        type: 'aresOEmbed',
        model: {
          oembed: {
            version: '1.0',
            title:
              'Cupcakes, You don watch my full interview for BBC Pidgin? How you see am? ðŸ™ŠðŸ’–âœ¨#BBCPidgin #CuppyOnAMission #CuppyxBBC',
            author_name: 'Cuppy',
            author_url: 'https://www.tiktok.com/@cuppymusic',
            provider_name: 'TikTok',
            provider_url: 'https://www.tiktok.com',
            thumbnail_url:
              'https://p16-sign-va.tiktokcdn.com/obj/tos-maliva-p-0068/328e9247ec2746b8992d5fcdde75affa_1649876928?x-expires=1663779600&x-signature=HP%2BukzBzbsK5budxvxLauElDen4%3D',
            thumbnail_width: 576,
            thumbnail_height: 1024,
            html: '<blockquote class="tiktok-embed" cite="https://www.tiktok.com/@cuppymusic/video/7086167423639997701" data-video-id="7086167423639997701" data-embed-from="oembed" style="max-width: 605px;min-width: 325px;" > <section> <a target="_blank" title="@cuppymusic" href="https://www.tiktok.com/@cuppymusic?refer=embed">@cuppymusic</a> <p>Cupcakes, You don watch my full interview for BBC Pidgin? How you see am? ðŸ™ŠðŸ’–âœ¨<a title="bbcpidgin" target="_blank" href="https://www.tiktok.com/tag/bbcpidgin?refer=embed">#BBCPidgin</a> <a title="cuppyonamission" target="_blank" href="https://www.tiktok.com/tag/cuppyonamission?refer=embed">#CuppyOnAMission</a> <a title="cuppyxbbc" target="_blank" href="https://www.tiktok.com/tag/cuppyxbbc?refer=embed">#CuppyxBBC</a></p> <a target="_blank" title="â™¬ original sound - Cuppy" href="https://www.tiktok.com/music/original-sound-7086167435610442501?refer=embed">â™¬ original sound - Cuppy</a> </section> </blockquote> <script async src="https://www.tiktok.com/embed.js"></script>',
          },
        },
      },
    ],
  },
};

export const facebookPostBlockEmbed = {
  type: 'renditions',
  model: {
    locator: 'urn:bbc:optimo:social:d4ae3e6f-51cf-4899-bbd4-16d14882cbc7',
    blocks: [
      {
        type: 'aresOEmbed',
        model: {
          oembed: {
            author_name: 'Rick Astley',
            author_url: 'https://www.facebook.com/RickAstley',
            provider_url: 'https://www.facebook.com',
            provider_name: 'Facebook',
            html: '<div id="fb-root"></div>\n<script async="1" defer="1" crossorigin="anonymous" src="https://connect.facebook.net/en_GB/sdk.js#xfbml=1&amp;version=v13.0" nonce="fPyyhrbz"></script><div class="fb-post" data-href="https://www.facebook.com/RickAstley/posts/545713756920775" data-width="552"><blockquote cite="https://graph.facebook.com/100044464658204/posts/545713756920775/" class="fb-xfbml-parse-ignore"><p>Had a fantastic time at &#064;thescriptofficial concert in LA this week, what a gig!! Have a great US tour guys! Donâ€™t miss...</p>Posted by <a href="https://www.facebook.com/RickAstley">Rick Astley</a> on&nbsp;<a href="https://graph.facebook.com/100044464658204/posts/545713756920775/">Friday, April 1, 2022</a></blockquote></div>',
            type: 'rich',
            version: '1.0',
            width: 552,
          },
        },
      },
    ],
  },
};

export const facebookVideoBlockEmbed = {
  type: 'renditions',
  model: {
    locator: 'urn:bbc:optimo:social:d4ae3e6f-51cf-4899-bbd4-16d14882cbc7',
    blocks: [
      {
        type: 'aresOEmbed',
        model: {
          oembed: {
            author_name: 'Rick Astley',
            author_url: 'https://www.facebook.com/RickAstley',
            provider_url: 'https://www.facebook.com',
            provider_name: 'Facebook',
            height: 888,
            html: '<div id="fb-root"></div>\n<script async="1" defer="1" crossorigin="anonymous" src="https://connect.facebook.net/en_GB/sdk.js#xfbml=1&amp;version=v13.0" nonce="8pqnGEmE"></script><div class="fb-video" data-href="https://www.facebook.com/RickAstley/videos/1378590239249667/"><blockquote cite="https://www.facebook.com/RickAstley/videos/1378590239249667/" class="fb-xfbml-parse-ignore"><a href="https://www.facebook.com/RickAstley/videos/1378590239249667/"></a><p>Saw another great gig in LA this week, I went to see Inhaler in the &#064;belascola and they were fantastic! Have a great show in New York tomorrow guys! Rick x &#064;inhalerdublin</p>Posted by <a href="https://www.facebook.com/RickAstley">Rick Astley</a> on Saturday, April 2, 2022</blockquote></div>',
            type: 'video',
            version: '1.0',
            width: 500,
          },
        },
      },
    ],
  },
};
