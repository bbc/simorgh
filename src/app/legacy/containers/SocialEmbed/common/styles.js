import styled from '@emotion/styled';
import { GEL_SPACING_TRPL } from '#psammead/gel-foundations/src/spacings';

/**
 * MAX_WIDTH        Ensures all embeds assume the same width. (Tweets max-out
 *                  at 500px, which is why this is set to 31.25rem.)
 * LAZYLOAD_OFFSET  The distance in pixels above or below the viewport before
 *                  an embed is allowed to load.
 */
const MAX_WIDTH = '31.25rem';
export const LAZYLOAD_OFFSET = 250;

const getWrapperHeightStyles = oEmbed => {
  /**
   * Adjust MIN_HEIGHT to configure the default minimum height of Social Embed
   * wrappers. This helps reduce layout shift. It is not applied to fallbacks.
   */
  const MIN_HEIGHT = '14rem';

  if (oEmbed?.height && oEmbed?.height !== '100%') {
    return `min-height: ${oEmbed.height / 16}rem`;
  }

  if (oEmbed) {
    return `min-height: ${MIN_HEIGHT};`;
  }
  return '';
};

export const Wrapper = styled.div`
  margin-bottom: ${GEL_SPACING_TRPL};
  max-width: ${MAX_WIDTH};
  ${({ oEmbed }) => getWrapperHeightStyles(oEmbed)}
`;
