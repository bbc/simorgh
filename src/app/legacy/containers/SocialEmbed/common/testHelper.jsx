import React from 'react';

import { RequestContextProvider } from '#contexts/RequestContext';
import { STORY_PAGE } from '#app/routes/utils/pageTypes';
import { ServiceContextProvider } from '../../../../contexts/ServiceContext';
import { ToggleContextProvider } from '../../../../contexts/ToggleContext';
import ThemeProvider from '../../../../components/ThemeProvider';

/**
 * withContexts is a higher-order component that returns a React component
 * wrapped in the necessary page contexts for testing and Stories.
 * @param {Function} Component A React component.
 * @param {Object} Configuration A configuration object.
 */
const withContexts =
  (Component, { isAmp, service = 'news', pageType = STORY_PAGE }) =>
  props =>
    (
      <RequestContextProvider
        isAmp={isAmp}
        pageType={pageType}
        service={service}
        pathname="/pathname"
      >
        <ThemeProvider service={service}>
          <ServiceContextProvider service={service}>
            <ToggleContextProvider
              toggles={{
                eventTracking: { enabled: false },
              }}
            >
              <Component {...props} />
            </ToggleContextProvider>
          </ServiceContextProvider>
        </ThemeProvider>
      </RequestContextProvider>
    );

export default withContexts;
