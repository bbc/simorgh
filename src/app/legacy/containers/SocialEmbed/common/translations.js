import pathOr from 'ramda/src/pathOr';

const createTranslations = ({ translations, index }) => {
  const skipLinkText = pathOr(
    'Skip %provider_name% content',
    ['socialEmbed', 'skipLink', 'text'],
    translations,
  ).concat(index > 0 ? `, ${index}` : '');

  const skipLinkEndTextVisuallyHidden = pathOr(
    'End of %provider_name% content',
    ['socialEmbed', 'skipLink', 'endTextVisuallyHidden'],
    translations,
  ).concat(index > 0 ? `, ${index}` : '');

  return {
    fallback: {
      text: pathOr(
        'Content is not available',
        ['socialEmbed', 'fallback', 'text'],
        translations,
      ),
      linkText: pathOr(
        'View content on %provider_name%',
        ['socialEmbed', 'fallback', 'linkText'],
        translations,
      ),
      linkTextSuffixVisuallyHidden: pathOr(
        ', external',
        ['socialEmbed', 'fallback', 'linkTextSuffixVisuallyHidden'],
        translations,
      ),
      warningText: pathOr(
        null,
        ['socialEmbed', 'fallback', 'warningText'],
        translations,
      ),
    },

    skipLink: {
      text: skipLinkText,
      endTextVisuallyHidden: skipLinkEndTextVisuallyHidden,
    },

    caption: {
      textPrefixVisuallyHidden: pathOr(
        'Video caption, ',
        ['socialEmbed', 'caption', 'textPrefixVisuallyHidden'],
        translations,
      ),
      text: pathOr(
        'Warning: Third party content may contain adverts',
        ['socialEmbed', 'caption', 'text'],
        translations,
      ),
      articleText: pathOr(
        'Warning: The BBC is not responsible for the content of external sites.',
        ['socialEmbed', 'caption', 'articleText'],
        translations,
      ),
      articleAdditionalText: pathOr(
        '%provider_name% content may contain adverts',
        ['socialEmbed', 'caption', 'articleAdditionalText'],
        translations,
      ),
    },
  };
};

export default createTranslations;
