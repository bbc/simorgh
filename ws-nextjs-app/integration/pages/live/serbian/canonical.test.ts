/**
 * @service serbian
 * @pathname /serbian/live/media-23179005/lat
 */

import { runCommonCrossPlatformTests } from '../../../../../src/integration/common';

describe('Canonical', () => {
  describe(pageType, () => {
    runCommonCrossPlatformTests();
  });
});
