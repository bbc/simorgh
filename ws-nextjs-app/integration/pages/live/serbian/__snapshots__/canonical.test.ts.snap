// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Canonical Live A11y I can see the skip to content link 1`] = `"Pređite na sadržaj"`;

exports[`Canonical Live Footer Anchors should match text and url 1`] = `
{
  "text": "BBC News, na srpskom",
  "url": "/serbian",
}
`;

exports[`Canonical Live Footer Anchors should match text and url 2`] = `
{
  "text": "Zašto BBC zaslužuje vaše poverenje",
  "url": "https://www.bbc.com/serbian/lat/institutional-50173730",
}
`;

exports[`Canonical Live Footer Anchors should match text and url 3`] = `
{
  "text": "Pravila korišćenja",
  "url": "https://www.bbc.com/terms",
}
`;

exports[`Canonical Live Footer Anchors should match text and url 4`] = `
{
  "text": "O BBC-ju",
  "url": "https://www.bbc.co.uk/aboutthebbc/",
}
`;

exports[`Canonical Live Footer Anchors should match text and url 5`] = `
{
  "text": "Pravila privatnosti",
  "url": "https://www.bbc.com/privacy/",
}
`;

exports[`Canonical Live Footer Anchors should match text and url 6`] = `
{
  "text": "Kolačići",
  "url": "https://www.bbc.com/usingthebbc/cookies/",
}
`;

exports[`Canonical Live Footer Anchors should match text and url 7`] = `
{
  "text": "Kontaktirajte BBC",
  "url": "https://www.bbc.co.uk/serbian/send/u50853665",
}
`;

exports[`Canonical Live Footer Anchors should match text and url 8`] = `
{
  "text": "Pročitajte naša pravila o linkovanju drugih sajtova.",
  "url": "https://www.bbc.co.uk/editorialguidelines/guidance/feeds-and-links",
}
`;

exports[`Canonical Live Footer I can click on the BBC branding and it would take me to the homepage 1`] = `"/serbian"`;

exports[`Canonical Live Footer Paragraphs should match text 1`] = `"© 2024 BBC. BBC nije odgovoran za sadržaj drugih sajtova. Pročitajte naša pravila o linkovanju drugih sajtova."`;

exports[`Canonical Live Header I can see the branding 1`] = `"BBC News, na srpskom"`;

exports[`Canonical Live Header Navigation link should match text and url 1`] = `
{
  "text": "Početna strana",
  "url": "/serbian/lat",
}
`;

exports[`Canonical Live Header Navigation link should match text and url 2`] = `
{
  "text": "Američki izbori",
  "url": "/serbian/lat/topics/c1v8enng1npt",
}
`;

exports[`Canonical Live Header Navigation link should match text and url 3`] = `
{
  "text": "Srbija",
  "url": "/serbian/lat/topics/cr50vdy9q6wt",
}
`;

exports[`Canonical Live Header Navigation link should match text and url 4`] = `
{
  "text": "Balkan",
  "url": "/serbian/lat/topics/c06g87137jgt",
}
`;

exports[`Canonical Live Header Navigation link should match text and url 5`] = `
{
  "text": "Svet",
  "url": "/serbian/lat/topics/c2lej05e1eqt",
}
`;

exports[`Canonical Live Header Navigation link should match text and url 6`] = `
{
  "text": "Video",
  "url": "/serbian/lat/topics/c44vyp5g049t",
}
`;

exports[`Canonical Live Header Navigation link should match text and url 7`] = `
{
  "text": "Najpopularnije",
  "url": "/serbian/lat/popular/read",
}
`;

exports[`Canonical Live Main heading should match text 1`] = `"Live test page"`;

exports[`Canonical Live SEO Apple Touch Icon should match attributes 1`] = `
{
  "sizes": null,
  "url": "http://localhost:7080/serbian/images/icons/icon-192x192.png",
}
`;

exports[`Canonical Live SEO Apple Touch Icon should match attributes 2`] = `
{
  "sizes": "72x72",
  "url": "https://static.files.bbci.co.uk/ws/simorgh-assets/public/serbian/images/icons/icon-72x72.png",
}
`;

exports[`Canonical Live SEO Apple Touch Icon should match attributes 3`] = `
{
  "sizes": "96x96",
  "url": "https://static.files.bbci.co.uk/ws/simorgh-assets/public/serbian/images/icons/icon-96x96.png",
}
`;

exports[`Canonical Live SEO Apple Touch Icon should match attributes 4`] = `
{
  "sizes": "128x128",
  "url": "https://static.files.bbci.co.uk/ws/simorgh-assets/public/serbian/images/icons/icon-128x128.png",
}
`;

exports[`Canonical Live SEO Apple Touch Icon should match attributes 5`] = `
{
  "sizes": "144x144",
  "url": "https://static.files.bbci.co.uk/ws/simorgh-assets/public/serbian/images/icons/icon-144x144.png",
}
`;

exports[`Canonical Live SEO Apple Touch Icon should match attributes 6`] = `
{
  "sizes": "152x152",
  "url": "https://static.files.bbci.co.uk/ws/simorgh-assets/public/serbian/images/icons/icon-152x152.png",
}
`;

exports[`Canonical Live SEO Apple Touch Icon should match attributes 7`] = `
{
  "sizes": "180x180",
  "url": "https://static.files.bbci.co.uk/ws/simorgh-assets/public/serbian/images/icons/icon-180x180.png",
}
`;

exports[`Canonical Live SEO Apple Touch Icon should match attributes 8`] = `
{
  "sizes": "192x192",
  "url": "https://static.files.bbci.co.uk/ws/simorgh-assets/public/serbian/images/icons/icon-192x192.png",
}
`;

exports[`Canonical Live SEO Apple Touch Icon should match attributes 9`] = `
{
  "sizes": "384x384",
  "url": "https://static.files.bbci.co.uk/ws/simorgh-assets/public/serbian/images/icons/icon-384x384.png",
}
`;

exports[`Canonical Live SEO Apple Touch Icon should match attributes 10`] = `
{
  "sizes": "512x512",
  "url": "https://static.files.bbci.co.uk/ws/simorgh-assets/public/serbian/images/icons/icon-512x512.png",
}
`;

exports[`Canonical Live SEO Canonical link 1`] = `"http://localhost:7080/serbian/live/media-23179005/lat"`;

exports[`Canonical Live SEO Dir attribute 1`] = `"ltr"`;

exports[`Canonical Live SEO FB app ID 1`] = `"1609039196070050"`;

exports[`Canonical Live SEO Heading level 1 1`] = `"Live test page"`;

exports[`Canonical Live SEO Lang attribute 1`] = `"sr-latn"`;

exports[`Canonical Live SEO Linked data should match text 1`] = `
{
  "@context": "http://schema.org",
  "@graph": [
    {
      "@type": "NewsArticle",
      "author": {
        "@type": "NewsMediaOrganization",
        "logo": {
          "@type": "ImageObject",
          "height": 576,
          "url": "https://news.files.bbci.co.uk/ws/img/logos/og/serbian.png",
          "width": 1024,
        },
        "name": "BBC News na srpskom",
        "noBylinesPolicy": "https://www.bbc.com/serbian/lat/institutional-50173730#authorexpertise",
      },
      "dateModified": "2018-01-10T15:00:00.000Z",
      "datePublished": "2018-01-10T15:00:00.000Z",
      "headline": "Live test page",
      "image": {
        "@type": "ImageObject",
        "height": 576,
        "url": "https://news.files.bbci.co.uk/ws/img/logos/og/serbian.png",
        "width": 1024,
      },
      "inLanguage": {
        "@type": "Language",
        "alternateName": "sr-latn",
        "name": "Serbian",
      },
      "mainEntityOfPage": {
        "@id": "http://localhost:7080/serbian/live/media-23179005/lat",
        "@type": "WebPage",
        "name": "Live test page",
      },
      "publisher": {
        "@type": "NewsMediaOrganization",
        "logo": {
          "@type": "ImageObject",
          "height": 576,
          "url": "https://news.files.bbci.co.uk/ws/img/logos/og/serbian.png",
          "width": 1024,
        },
        "name": "BBC News na srpskom",
        "publishingPrinciples": "https://www.bbc.com/serbian/lat/institutional-50173730",
      },
      "thumbnailUrl": "https://news.files.bbci.co.uk/ws/img/logos/og/serbian.png",
      "url": "http://localhost:7080/serbian/live/media-23179005/lat",
    },
    {
      "@type": "LiveBlogPosting",
      "coverageEndTime": "2018-01-10T22:00:00.000Z",
      "coverageStartTime": "2018-01-10T15:00:00.000Z",
      "liveBlogUpdate": [],
    },
  ],
}
`;

exports[`Canonical Live SEO OG description 1`] = `"Live test page"`;

exports[`Canonical Live SEO OG image 1`] = `"https://news.files.bbci.co.uk/ws/img/logos/og/serbian.png"`;

exports[`Canonical Live SEO OG image alt 1`] = `"BBC News na srpskom"`;

exports[`Canonical Live SEO OG locale 1`] = `"sr-latn"`;

exports[`Canonical Live SEO OG site name 1`] = `"BBC News na srpskom"`;

exports[`Canonical Live SEO OG title 1`] = `"Live test page - BBC News na srpskom"`;

exports[`Canonical Live SEO OG type 1`] = `"website"`;

exports[`Canonical Live SEO OG url 1`] = `"http://localhost:7080/serbian/live/media-23179005/lat"`;

exports[`Canonical Live SEO Page title 1`] = `"Live test page - BBC News na srpskom"`;

exports[`Canonical Live SEO Robots meta tag 1`] = `"noodp, noydir, max-image-preview:large"`;

exports[`Canonical Live SEO Twitter card 1`] = `"summary_large_image"`;

exports[`Canonical Live SEO Twitter creator 1`] = `"@bbcnasrpskom"`;

exports[`Canonical Live SEO Twitter description 1`] = `"Live test page"`;

exports[`Canonical Live SEO Twitter image alt 1`] = `"BBC News na srpskom"`;

exports[`Canonical Live SEO Twitter image src 1`] = `"https://news.files.bbci.co.uk/ws/img/logos/og/serbian.png"`;

exports[`Canonical Live SEO Twitter site 1`] = `"@bbcnasrpskom"`;

exports[`Canonical Live SEO Twitter title 1`] = `"Live test page - BBC News na srpskom"`;
